src/core/scene.ts

```diff
@@ -1363,4 +1370,281 @@ export class Scene {
+      } else if (versionCompare(getVersion()).is.lessThan(scenePresetsVersion)) {
+        reject(Error('Not supported in this XBC version'));
+      } else {
+        iApp.get('scenepreset:' + this._uid).then(value => {
+          resolve(value);
+        });
+      }
+    });
+  }
+
+  /**
+   * param: (preset: string)
+   * ```
+   * return: Promise<boolean>
+   * ```
+   * Switch to the specified preset for the scene
+   *
+   * #### Usage
+   *
+   * ```javascript
+   *
+   * myScene.getPresets()
+   * .then(presets => {
+   *   const lastPreset = presets.pop()
+   *   return myScene.switchToPreset(lastPreset);
+   * })
+   * .then(isSwitched => {
+   *   console.log('switched to preset : ' + isSwitched)
+   * });
+   * ```
+   */
+  switchToPreset(preset: string): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      if (Environment.isSourcePlugin()) {
+        reject(Error('Not supported on source plugins'));
+      } else if (versionCompare(getVersion()).is.lessThan(scenePresetsVersion)) {
+        reject(Error('Not supported in this XBC version'));
+      } else {
+        iApp.set('scenepreset:' + this._uid, preset).then(value => {
+          resolve(value);
+        });
+      }
+    })
+  }
+
+  /**
+   * return: Promise<string>
+   *
+   * Add a new preset to the scene, returns the UID of the new preset
+   * Does not work on source plugins.
+   *
+   * #### Usage
+   *
+   * ```javascript
+   * myScene.addPreset().then(function(preset) {
+   *  console.log('New preset UID is ' + preset);
+   * });
+   * ```
+   */
+  addPreset(): Promise<string> {
+    return new Promise((resolve, reject) => {
+      if (Environment.isSourcePlugin()) {
+        reject(Error('Not supported on source plugins'));
+      } else if (versionCompare(getVersion()).is.lessThan(scenePresetsVersion)) {
+        reject(Error('Not supported in this XBC version'));
+      } else {
+        iApp.get('scenenewpreset:' + this._uid).then(value => {
+          resolve(value);
+        });
+      }
+    });
+  }
+
+  /**
+   * param: (preset: string)
+   * ```
+   * return: Promise<boolean>
+   * ```
+   * Remove the specified preset for the scene
+   *
+   * #### Usage
+   *
+   * ```javascript
+   *
+   * myScene.removePreset(lastPreset)
+   * .then(isRemoved => {
+   *   console.log('preset is removed : ' + isRemoved)
+   * });
+   * ```
+   */
+  removePreset(preset: string): Promise<boolean> {
+    return new Promise((resolve, reject) => {
+      if (Environment.isSourcePlugin()) {
+        reject(Error('Not supported on source plugins'));
+      } else if (versionCompare(getVersion()).is.lessThan(scenePresetsVersion)) {
+        reject(Error('Not supported in this XBC version'));
+      } else if (preset === '{00000000-0000-0000-0000-000000000000}') {
+        reject(Error('Cannot delete the default preset'));
+      } else {
+        iApp.set('sceneremovepreset:' + this._uid, preset).then(value => {
```

Comment:
This returns false if setting a non-existent preset. Do you think it's better if we reject it?

Reply:
Yeah, I assume that setting the preset as the default preset would also resolve with a `false` right? We're calling `reject` for that one, which I agree on as it is always better that the developer would receive an error message in the remote debugger rather than failing silently, since it's most likely a bug on their code rather than something intentional.

Reply:
Again, my bad, *deleting* a non-existent preset. Makes sense, will reject if the set/remove method returns false.